### PR TITLE
MODEMAIL-15 Improve 'mod-email' performance

### PR DESCRIPTION
**Pull request by OleksiiMaksymov**
**Description**
1. Add the ability to cache and force reset SMTP configuration
2. Provide a temporary buffer before sending email

**Link**
[MODEMAIL-15](https://issues.folio.org/browse/MODEMAIL-15)

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -25,18 +25,132 @@
...
   ],
   "permissionSets": [
     {
       "permissionName": "email.message.post",
       "displayName": "message - send email notifications",
       "description": "send email notifications"
+    },
+    {
+      "permissionName": "email.message.collection.get",
+      "displayName": "get email messages ",
```
Comment (kostyantyn-kh) [inline descriptors/ModuleDescriptor-template.json]: Please remove extra space

File: ramls/email_entity.json [hunk 2]
```diff
@@ -28,6 +32,16 @@
       "description":"text of email",
       "type": "string"
     },
+    "status": {
+      "description":"status of email",
+      "type": "string",
+      "default": "NEW"
```
Comment (kostyantyn-kh) [inline ramls/email_entity.json]: Is it worth using enum here?

File: src/main/java/org/folio/enums/SendingStatus.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package org.folio.enums;
+
+import java.util.Arrays;
+
+public enum SendingStatus {
+  NEW,
+  IN_PROCESS,
```
Comment (kostyantyn-kh) [inline src/main/java/org/folio/enums/SendingStatus.java]: PROCESSING?

Comment (OleksiiMaksymov) [reply src/main/java/org/folio/enums/SendingStatus.java]: done

File: ramls/email_entity.json (hunk 2 above)
Comment (OleksiiMaksymov) [reply ramls/email_entity.json]: done

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (OleksiiMaksymov) [reply descriptors/ModuleDescriptor-template.json]: fix

**Comment by OleksiiMaksymov** [comment]
The current implementation will be replaced by the publisher-subscriber implementation.

**Pull request closed without merging**